pages/search.py

```diff
@@ -1,11 +1,12 @@
 from selenium.webdriver.common.by import By
+from pages.base_page import BasePage
 
-class Search:
+class Search(BasePage):
     def __init__(self, driver):
-        self.driver = driver
+        super().__init__(driver)
         self.search_box = (By.ID, "twotabsearchtextbox")
         self.search_button = (By.ID, "nav-search-submit-button")
-
+    
     def search(self, query):
-        self.driver.find_element(*self.search_box).send_keys(query)
-        self.driver.find_element(*self.search_button).click()
+        BasePage.wait_for_element(self.search_box).send_keys(query)
+        BasePage.wait_for_element(self.search_button).click()
```

Comment:
no hace falta mencionar a BasePage.nombre_metodo, deberia ser suficiente con "self". Ademas tené en cuenta que generaste un método para no tener que repetir el wait_for_element:
```suggestion
        self.click(self.search_button)
```

Reply:
@LorenzoQuintana